app/src/services/bucket.js

```diff
@@ -7,6 +7,38 @@ const { Bucket } = require('../db/models');
  * The Bucket DB Service
  */
 const service = {
+
+  /**
+   * @function checkBucketBasicAccess
+   * Grants a user provided permissions to the bucket if the data precisely matches
+   * accessKeyId, bucketId, bucket and endpoint values.
+   * @param {string} data.bucketId The COMS guid for the bucket
+   * @param {string} data.bucket The S3 bucket name
+   * @param {string} data.endpoint The S3 bucket endpoint
+   * @param {string} data.accessKeyId The S3 bucket secret access key
+   * @param {object} [etrx=undefined] An optional Objection Transaction object
+   * @returns {Promise<object>} The result will include all bucketIds that pass the validation
+   * @throws The error encountered upon db transaction failure
+   */
+  checkBucketBasicAccess: async (data, etrx = undefined) => {
```

Comment:
maybe consider using the existing `bucketService.readUnique()` function to reduce dupication
the result could be modified after back in the authorization.js

Reply:
actually maybe not. nothing wrong with having a dedicated function for clarity